cpp/memilio/io/binary_serializer.h

```diff
@@ -248,34 +345,45 @@ template <class T, std::enable_if_t<std::is_trivial<T>::value, void*>>
 IOResult<T> BinarySerializerObject::expect_element(const std::string& name, Tag<T> /*tag*/)
 {
     mio::unused(name);
-    T t;
-    if (m_stream.read(reinterpret_cast<unsigned char*>(std::addressof(t)), sizeof(t))) {
-        return mio::success(t);
-    }
-    else {
-        return mio::failure(mio::StatusCode::UnknownError, "Unexpected EOF reading " + name + " from binary stream.");
+
+    if (*m_status) {
```

Comment:
It's similar to our discussions about `BinaryStream::read` above, does true mean success or failure? I guess it's a good idea to be explicit if possible. In this case it would be m_status->is_ok() or something.

Also, I noticed that it's easy to get the check wrong entirely because m_status is a pointer. `if (m_status)` would only check the pointer for null. Probably safer to avoid automatic bool conversions...